qiskit_cold_atom/fermions/ffsim_backend.py

```diff
@@ -0,0 +1,388 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Fermionic simulator backend that uses ffsim."""
+
+from __future__ import annotations
+
+import datetime
+import time
+import uuid
+import warnings
+from collections import Counter
+from typing import Any, Dict, List, Optional, Union
+
+import ffsim  # pylint: disable=import-error
+import numpy as np
+import scipy.linalg
+from qiskit import QuantumCircuit
+from qiskit.providers import Options
+from qiskit.providers.models import BackendConfiguration
+from qiskit.result import Result
+from qiskit_aer import AerJob
+from qiskit_nature.operators.second_quantization import FermionicOp
+from scipy.sparse.linalg import expm_multiply
+
+from qiskit_cold_atom.circuit_tools import CircuitTools
+from qiskit_cold_atom.fermions.base_fermion_backend import BaseFermionBackend
+from qiskit_cold_atom.fermions.fermion_gate_library import (
+    FermionicGate,
+    Hop,
+    Interaction,
+    LoadFermions,
+    Phase,
+)
+
+
+class FfsimBackend(BaseFermionBackend):
+    """Fermionic simulator backend that uses ffsim.
+
+    This is a specialized simulator backend for fermionic circuits that uses `ffsim`_.
+    It computes the statevector and simulate measurements with vastly improved efficiency
+    compared with the :class:`~.FermionSimulator` backend. However, unlike
+    :class:`~.FermionSimulator`,
+    it does not compute the full unitary of a circuit.
+
+    This backend is not supported on Windows, and a special ``pip`` command is needed
+    to install it:
+
+    .. code::
+
+        pip install "qiskit-cold-atom[ffsim]"
+
+    .. _ffsim: https://github.com/qiskit-community/ffsim
+    """
+
+    _DEFAULT_CONFIGURATION = {
+        "backend_name": "ffsim_simulator",
+        "backend_version": "0.0.1",
+        "n_qubits": 100,
+        "basis_gates": None,
```

Comment:
You can / should most likely also add the `num_species` argument here. Makes it clear that two species are required. 

Same for the keyword `cold_atom_type` which should be `fermion`

Reply:
I simply copied this from the existing [FermionSimulator](https://github.com/qiskit-community/qiskit-cold-atom/blob/0e2396f88009a370dbd25c9d029bd81859722af2/qiskit_cold_atom/fermions/fermion_simulator_backend.py#L40) backend. Does that need to be updated as well?

I don't think `num_species` needs to be set as part of the configuration, since it is passed as an argument to `run`. Also, I've updated this PR to add support for 1 species as well.

Reply:
@LaurinFischer I must admit that I also observed that this is missing in original fermion simulator. Is this by design of was this simply a "missing" feature for which we might an issue ?

Reply:
Hi, this is by design, yes. The thought was that the `FermionSimulator` is supposed to be a generic fermionic backend that itself imposes no specific number of species to keep maximal flexibility. Seems like this is also the case for the ffsim backend now, so I would stick to that. 
In contrast, a backend that describes an actual experiment with a fixed number of species, like the `FermionicTweezerSimulator`, should define this number in the backend configuration.